esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2912,17 +2913,11 @@ bool CFileSprayEx::onFileList(IEspContext &context, IEspFileListRequest &req, IE
         }
         else
         {
+            StringBuffer dzName;
             if (isEmptyString(dropZoneName))
-            {
-                //Call findDropZonePlane() to resolve plane by hostname. Shouldn't resolve plane
-                //by hostname in containerized but kept for backward compatibility for now.
-                Owned<IPropertyTree> plane = findDropZonePlane(sPath, netaddr, true, true);
-                dropZoneName = plane->queryProp("@name");
-            }
-            SecAccessFlags permission = getDZPathScopePermissions(context, dropZoneName, sPath, nullptr);
-            if (permission < SecAccess_Read)
-                throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Access DropZone Scope %s %s not allowed for user %s (permission:%s). Read Access Required.",
-                    dropZoneName, sPath.str(), context.queryUserId(), getSecAccessFlagName(permission));
+                dropZoneName = findDropZonePlaneName(netaddr, sPath, dzName);
+            if (!isEmptyString(dropZoneName))
+                checkDZPathScopePermissions(context, dropZoneName, sPath, SecAccess_Read);
```

Comment:
before checking LDAP, I think it should 1st validate that the path is within the named dropzone.
If there's not an existing function that checks if a path is valid for a plane(?), one should be added to dautils.?pp

Reply:
maybe beyond the scope of this PR, but the above code and use of legacy method validateDropZonePath (lines 2935-2940) should be revised (code like validateDropZonePath should be deleted).

I think this code section should look something like this:
```
        else
        {
            StringBuffer dzName;
            if (isEmptyString(dropZoneName))
                dropZoneName = findDropZonePlaneName(netaddr, sPath, dzName);
            Owned<IPropertyTree> dropZone;
            if (!isEmptyString(dropZoneName)) // must be true, unless bare-metal and isDropZoneRestrictionEnabled()==false
            {
                dropZone.setown(getDropZonePlane(dropZoneName));
                if (!dropZone)
                    throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Unknown landing zone: %s", dropZoneName);
                if (!isPathInDropZone(dropZone, sPath))
                    throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Path '%s' is not valid for dropzone '%s'", sPath.str(), dropZoneName);

                checkDZPathScopePermissions(context, dropZoneName, sPath, SecAccess_Read);
            }

            StringArray hosts;
            if (!isEmptyString(netaddr))
            {
                if (dropZone && !isHostInPlane(dropZone, netaddr, false))
                    throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Host '%s' is not valid for dropzone '%s'", netaddr, dropZoneName);

                hosts.append(netaddr);
            }
            else
            {
                // NB: if no netaddr, a dropZone must have been specified
                getPlaneHosts(hosts, dropZone);
                if (!hosts.ordinality())
                    hosts.append("localhost");
            }

            ForEachItemIn(i, hosts)
            {
                const char* host = hosts.item(i);
                getPhysicalFiles(context, dropZoneName, host, sPath, fileNameMask, directoryOnly, files);
            }
        }
```

Reply:
The validateDropZonePath is also used in ws_fsBinding.cpp. I guess I should do the same (if there is dropzoneName, calls isPathInDropZone()) there.

Reply:
I made the changes. Works for bare-metal. In my local Docker Desktop, the code failed in the isHostInPlane(dropZone, netaddr, false): Host 'localhost' is not valid for dropzone 'mydropzone'

Reply:
Inside the isHostInPlane(), if I change the line 62 to: 
return streq(host, "localhost") ? true : false;
it seems working. I am not sure we should add that change. @jakesmith please advise.

Reply:
>return streq(host, "localhost") ? true : false;

I don't think that doesn't sound right.

Where is "localhost" coming from in this context?